### ISSUE TITLE
Cargo warning when you try to publish a crate that contains files/folders with windows reserved filenames

### ISSUE THREAD

**Issue by CPerezz** [rust-lang/cargo#8163]
**Problem**
Actually, you can create a lib or a bin where the package contains files such as `aux.rs` which have Windows reserved filenames.
As soon as #8055 is closed, cargo will display a proper error when you try to unpack these types of packages.

But it actually doesn't say anything to the user when you try to build/publish a crate.
Obviously, you maybe are not building your package to be ran in Windows but Linux, and then this is not an issue at all. 

But maybe it should at least make the user know that if he is targeting Windows as an OS where the package is going to be deployed, using windows reserved filenames will make the package impossible to install or run.

**Possible Solution(s)**
- Add a warning when you're trying to publish a crate with reserved filenames on any of it's files or folders. Nothing that prevents you to do this, but that explains that this can cause errors as the one that will be printed once #8055 lands.
- Imo it will be so annoying to have a warning on build level. Even something like `#![allow(windows_reserved_filenames)]` will do the job. But again, seems too much to me.

I don't know if that makes sense from the teams perspective, but the previous issue made me think that if we will error when we find that filenames, we should then advise users before publishing.

**Comment by ehuss** [comment]
It should display a warning on all platforms like this:

```
warning: file src/aux.rs is a reserved Windows filename, it will not work on Windows platforms
```

Are you not seeing that on the latest nightly?

**Comment by CPerezz** [comment]
I'm not seeing it on the latest stable. I just tried, that's why I opened the issue.

Is there any reason why it is only triggered in nightly?

**Comment by ehuss** [comment]
It was only implemented a few weeks ago in #7959. It takes anywhere from 6 to 12 weeks for changes to make it to stable. You can also check the [`CHANGELOG`](https://github.com/rust-lang/cargo/blob/master/CHANGELOG.md), which is updated once every 6 weeks.

**Comment by CPerezz** [comment]
Thanks @ehuss my bad, I should have reviewed it!

Sorry for the inconvenience then :smile:
